### PR TITLE
add skill

### PR DESCRIPTION
**Pull request by NguyenHai2310**

### CODE REVIEW

File: app/controllers/admin/skills_controller.rb [hunk 1]
```diff
@@ -20,13 +20,20 @@ def edit
...
       redirect_to admin_skills_path
     else
       render :edit
     end
   end
 
+  def destroy
+    @skill = Skill.find params[:id]
+    @skill.destroy
+    flash[:success] = t "skill.deleted"
```
Comment (vjnhat) [inline app/controllers/admin/skills_controller.rb]: you should change it look like
if success flash[:success] else flash[:fail]
redirect_to things

File: app/controllers/skill_users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class SkillUsersController < ApplicationController
+  def show
+    @user = User.find params[:user_id]
+    @skills = Skill.all
+    @skills.each do |skill|
+      unless @user.skills.include? skill
+        @user.skill_users.build skill_id: skill.id
```
Comment (khanhkhanh) [inline app/controllers/skill_users_controller.rb]: What do you mean here?

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<% provide :title, @user.username %>
+<%= link_to "edit", user_skills_path(@user) %>
```
Comment (khanhkhanh) [inline app/views/users/show.html.erb]: i18n

File: app/views/skill_users/_skill_form.html.erb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+      <tr>
+        <th><%= t "skill.name" %><th>
+        <th><%= t "skill.level" %></th>
+        <th><%= t "skill.year" %></th>
+      </tr>
+      <tr>
+       <%= f.fields_for :skill_users do |builder| %>
+            <tr>
+              <td><%= builder.check_box :_destroy,
+               {checked: @user.skills.include?(builder.object.skill)},Settings.zero,Settings.one %></td>
```
Comment (khanhkhanh) [inline app/views/skill_users/_skill_form.html.erb]: Need spaces before Settings

File: app/models/user.rb [hunk 5]
```diff
@@ -4,6 +4,11 @@ class User < ActiveRecord::Base
...
          :recoverable, :rememberable, :trackable, :validatable
 
+  has_many :skill_users, dependent: :destroy
+  has_many :skills, through: :skill_users
+
+  accepts_nested_attributes_for :skill_users, allow_destroy: true
+
   validates :username, presence: true, uniqueness: true,
     length: {maximum: Settings.user.username.maximum}
   validates :email, presence: true,
```
Comment (khanhkhanh) [inline app/models/user.rb]: Remove ","

File: app/controllers/skill_users_controller.rb (hunk 2 above)
Comment (NguyenHai2310) [inline app/controllers/skill_users_controller.rb]: cann't remove
ìf remove ,error will happen

File: app/controllers/admin/skills_controller.rb [hunk 6]
```diff
@@ -20,13 +20,24 @@ def edit
...
+      flash[:success] = t "skill.updated"
       redirect_to admin_skills_path
     else
       render :edit
     end
   end
 
+  def destroy
+    @skill = Skill.find params[:id]
+    if @skill.destroy
```
Comment (vjnhat) [inline app/controllers/admin/skills_controller.rb]: you should change it like that
if @skill.destroy
  flash[:success] = t "skill.deleted"
else
  flash[:danger] = t "skill.fail"
end
redirect_to admin_skills_path

File: app/models/skill_user.rb [hunk 7]
```diff
@@ -0,0 +1,6 @@
+class SkillUser < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :skill
+
+  validates :skill, presence: true
+end
```
Comment (vjnhat) [inline app/models/skill_user.rb]: new line
let remember this error, you can see it easily  when review your code

File: app/views/skill_users/show.html.erb [hunk 8]
```diff
@@ -0,0 +1,14 @@
+<% provide :title, t("skilluser.add") %>
+<h1><%= t "skilluser.add" %></h1>
+<%= link_to " ", user_path(current_user), class: "glyphicon glyphicon-menu-left" %>
+<div class="row">
+  <div class= "col-md-8">
+    <%= render "skill_form" %>
+  </div>
+  <div class= "col-md-4">
+    <div class= "list-skill">
+      <h3><%= t "skill.name" %><h3>
```
Comment (vjnhat) [inline app/views/skill_users/show.html.erb]: you forgot </h3>, didn't you???

File: app/views/admin/skills/new.html.erb [hunk 9]
```diff
@@ -1,3 +1,3 @@
-<% provide :title, t("skill.new") %>
-<h1><%= t("skill.new") %></h1>
+<% provide :title, t("skill.create") %>
+<h1><%= t "skill.new" %></h1>
 <%= render "form_skill" %>
```
Comment (vjnhat) [inline app/views/admin/skills/new.html.erb]: let remember this misstake,you can see it easily when review your code. 
you have to add a new line in end file.

File: app/views/skill_users/show.html.erb [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+<%= link_to user_path(current_user) do %>
+  <span class="glyphicon glyphicon-backward"></span>
+<% end %>
+<div class="row">
+  <div class= "col-md-8">
+    <%= render "skill_form" %>
+  </div>
+  <div class= "col-md-4">
+    <div class= "list-skill">
+      <h3><%= t "skill.name" %><h3>
```
Comment (vjnhat) [inline app/views/skill_users/show.html.erb]: </h3> ?

File: app/models/user.rb (hunk 5 above)
Comment (NguyenHai2310) [inline app/models/user.rb]: chỗ này bên dưới có  phần length, nên không xoa được ","

Comment (brokenheart1712) [inline app/models/user.rb]: Không bỏ được dấu phẩy ở đấy, còn dòng dưới cơ mà

File: app/controllers/admin/skills_controller.rb [hunk 11]
```diff
@@ -8,7 +8,7 @@ def create
     if @skill.save
       flash[:success] = t("skill.add")
```
Comment (phonghenry14) [inline app/controllers/admin/skills_controller.rb]: remove ()

File: config/locales/en.yml [hunk 12]
```diff
@@ -75,3 +82,14 @@ en:
     type: "Type:"
   form:
     error_contain: "the form contains"
+=======
```
Comment (phonghenry14) [inline config/locales/en.yml]: remove

**Pull request merged**